kube/src/service/mod.rs

```diff
@@ -84,6 +86,7 @@ impl TryFrom<Config> for Service {
             .map_request(move |r| set_default_headers(r, default_headers.clone()))
             .map_request(accept_compressed)
             .map_response(maybe_decompress)
+            .layer(AuthLayer::new(auth))
```

Comment:
We don't need to use this layer unless `Authentication::RefreshableToken(_)`. `Basic` and `Token` can use `default_headers` instead.

Reply:
maybe we should bundle it in with the `oauth` feature in that case. It's just a local development option anyway.

Reply:
think i'll just put out a call for help on the azure provider, so we can get some data on how composable the provider logic is. though i doubt this is high priority for most people writing controllers, as you can [impersonate a service account](https://clux.github.io/probes/post/2019-03-31-impersonating-kube-accounts/) if you have good access in a cluster.

Reply:
Opened https://github.com/clux/kube-rs/pull/396 to use it conditionally.